src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csip/CSIPResource.kt

```diff
@@ -0,0 +1,211 @@
+    description = "Gets csip details. Requires role NOMIS_CSIP",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Invalid request",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires role NOMIS_CSIP",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Not found",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+    ],
+  )
+  fun getCSIP(
+    @Schema(description = "CSIP id") @PathVariable id: Long,
+  ) = csipService.getCSIP(id)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_CSIP')")
+  @GetMapping("/count")
+  @Operation(
+    summary = "Get csip count",
+    description = "Gets a count of all csips. Requires role NOMIS_CSIP",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires role NOMIS_CSIP",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+    ],
+  )
+  fun getCSIPCount() = csipService.getCSIPCount()
+}
+
+@Schema(description = "CSIP Details")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class CSIPResponse(
+  @Schema(description = "The csip id")
+  val id: Long,
+  @Schema(description = "The id of the offender")
+  val offenderNo: String,
+  @Schema(description = "The booking id associated with the CSIP")
+  val bookingId: Long?,
+
+  @Schema(description = "Type of incident")
+  val type: CodeDescription,
+  @Schema(description = "Location of the incident")
+  val location: CodeDescription,
+  @Schema(description = "The Area of work where the incident took place aka function")
+  val areaOfWork: CodeDescription,
+
+  @Schema(description = "CSIP Plans")
+  val plans: List<Plan>,
+)
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class Staff(
```

Comment:
I couldn't see where Staff and Offender is used

Reply:
yet to come